assignments/assignment02.rb

```diff
@@ -0,0 +1,187 @@
+# ========================================================================================
+#  Assignment submitted by Gordon Turibamwe
+#
+# ========================================================================================
+# Assignment 2
+# ========================================================================================
+
+# ========================================================================================
+#  Problem 1 - `to_sentence`
+
+# implement method `to_sentence`
+
+# creates an english string from array
+
+def to_sentence(ary)  
+  # PSEUDO CODE
+  # get array
+  # create new string
+  # add each word to new string
+  # if more than 2 words, add "and" before the last word
+  # print sentence
+
+  sentence = ""
+  ary.each do |word|
+    if ary.length > 2 && ary.index(word) == ary.length - 1
+      sentence += "and "
+    end
+    sentence += word.to_s + ',' + " "
+  end
+  sentence = sentence.chomp(", ")
+print sentence
+return sentence #added this tests
+end
+
+# Your method should generate the following results:
+#to_sentence []                       #=> ""
+#to_sentence ["john"]                 #=> "john"
+#to_sentence ["john", "paul"]         #=> "john and paul"
+#to_sentence [1, "paul", 3, "ringo"]  #=> "1, paul, 3 and ringo"
+
+# ========================================================================================
+#  Problem 2 - `mean, median`
+
+# implement methods "mean", "median" on Array of numbers
+def mean(ary)
+  # PSEUDO CODE
```

Comment:
there are more elegant ways in ruby to sum an array of values than iterating over them one-by-one. I showed in the lecture how you can use reduce to sum an array of values.